crates/astria-sequencer/src/app.rs

```diff
@@ -2111,6 +2182,52 @@ mod test {
         );
     }
 
+    #[tokio::test]
+    async fn app_deliver_tx_invalid_chain_id() {
```

Comment:
move/change this test to test the mempool check (after you delete the check_stateful check this test will fail)